### ISSUE TITLE
TextBundle does not recognize it is a child of a NodeBundle

### ISSUE THREAD

**Issue by campbellcole** [bevyengine/bevy#6959]
## Bevy version

`0.9.1`

## What you did

Spawning the following two entities causes the console to be flooded with warning messages that appear to be incorrect:
```rs
commands
    .spawn((
        NodeBundle {
            style: Style {
                size: Size::AUTO,
                margin: UiRect::all(Val::Auto),
                ..default()
            },
            ..default()
        },
        MusicDropUI,
    ))
    .with_children(|parent| {
        parent.spawn(
            TextBundle::from_section(
                "Drag and drop a music file to begin",
                TextStyle {
                    font: font_handles.fonts.get(&TryptFont::Regular).unwrap().clone(),
                    font_size: 32.0,
                    color: Color::WHITE,
                },
            )
            .with_style(Style {
                align_self: AlignSelf::Center,
                ..default()
            }),
        );
    });
```

Some context, as I imagine this may not be reproducible from just this code: This setup function runs immediately after despawning every entity in the world, so it should be a fresh start (and using `bevy_inspector_egui` correctly demonstrates this). The only two existing entities in the world after this command are a `Camera2DBundle` and the root UI node, with it's `TextBundle` child:

![image](https://user-images.githubusercontent.com/10430178/207771431-480601b1-d9c3-49ec-9dd3-38edf64f61d7.png)

## What went wrong

The console gets flooded with warnings stating that a style component has a parent which is not style, but this is not the case. The only entity with a parent has a ui node as it's parent.

## Additional information

This message only appears after despawning a previous UI hierarchy and then spawning this one. I cannot figure out how to make a minimum reproducible example because I have no idea what is causing this to happen.

**Comment by nicopap** [comment]
I can't reproduce with the minimal example you provided. Is there other code than what you posted? Could I take a look at it?

**Comment by campbellcole** [comment]
I just figured out what the problem was. I was using a helper system that I labeled `despawn_recursive`, but somehow I managed to use the `despawn` function instead of `despawn_recursive`. This was creating orphan UI entities that did not appear in the inspector GUI.

To show how this happened, first we have the function responsible for registering these systems with the app:

```rs
app.add_enter_system_set(
    AppState::WaitingForMusic,
    SystemSet::new()
        .with_system(utils::despawn_recursive::<MenuComponent>)
        .with_system(setup_dnd_ui),
)
.add_system(wait_for_drop.run_in_state(AppState::WaitingForMusic));
```

You can see my helper system `despawn_recursive<C: Component>` being used here, however the definition was (erroneously) as follows:

```rs
pub fn despawn_recursive<T: Component>(mut commands: Commands, query: Query<Entity, With<T>>) {
    for entity in query.iter() {
        commands.entity(entity).despawn();
    }
}
```

Changing this to `commands.entity(entity).despawn_recursive()` solved the problem. I don't know how I messed this up, but I'm glad I figured it out. I do wonder though, and perhaps this is out of scope for this crate, but is there any defined behavior for orphaning an entity? It seems like the child entity still believes it has a parent, but since the parent entity was deleted it finds nothing, including not finding a Node component. My first guess would be that the child becomes the new parent, but that doesn't seem to be happening. Thanks for the quick reply btw.

**Comment by nicopap** [comment]
It is a bug to have an entity with a `Parent(foo)` component when `foo` doesn't have a child. I think it is also a bug to have a `Parent(v)` component where `v` is an empty list. If you can reproduce this, could you share the code to reproduce it? Then I could open a new issue.

**Comment by mockersf** [comment]
`despawn` has no knowledge of hierarchy and so will not update `Parent` or `Children` components by design.

**Comment by Weibye** [comment]
Should this be closed as "working as intended" or renamed to "Bug: `Despawn` does not update Parent / Children components"?

**Comment by rparrett** [comment]
At the very least this could be closed as a duplicate of #5584.
